pkg/broker/handler/pool.go

```diff
@@ -80,6 +84,9 @@ func (c *probeChecker) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
+	// Perform Authentication check.
+	authcheck.AuthenticationCheck(req.Context(), c.logger, c.authType, w)
```

Comment:
Should this return under some condition? Such as if the auth check fails.

Reply:
It only write statue into the response, which I included in the authcheck.AuthenticationCheck previously. Based on your comment, I'll do:
```
if err := authcheck.AuthenticationCheck; err != nil {
			pc.logger.Error("authentication check failed", zap.Error(err))
			w.WriteHeader(http.StatusUnauthorized)
                         return
}
```